src/extractor/extractor.cpp

```diff
@@ -317,6 +191,140 @@ int Extractor::run(ScriptingEnvironment &scripting_environment)
     return 0;
 }
 
+std::vector<TurnRestriction>
+Extractor::ParseOSMData(ScriptingEnvironment &scripting_environment,
+                        const unsigned number_of_threads)
+{
+    TIMER_START(extracting);
+
+    util::Log() << "Input file: " << config.input_path.filename().string();
+    if (!config.profile_path.empty())
+    {
+        util::Log() << "Profile: " << config.profile_path.filename().string();
+    }
+    util::Log() << "Threads: " << number_of_threads;
+
+    const osmium::io::File input_file(config.input_path.string());
+
+    osmium::io::Reader reader(
+        input_file, (config.use_metadata ? osmium::io::read_meta::yes : osmium::io::read_meta::no));
+
+    const osmium::io::Header header = reader.header();
+
+    unsigned number_of_nodes = 0;
+    unsigned number_of_ways = 0;
+    unsigned number_of_relations = 0;
+
+    util::Log() << "Parsing in progress..";
+    TIMER_START(parsing);
+
+    ExtractionContainers extraction_containers;
+    auto extractor_callbacks = std::make_unique<ExtractorCallbacks>(
+        extraction_containers, scripting_environment.GetProfileProperties());
+
+    // setup raster sources
+    scripting_environment.SetupSources();
+
+    std::string generator = header.get("generator");
+    if (generator.empty())
+    {
+        generator = "unknown tool";
+    }
+    util::Log() << "input file generated by " << generator;
+
+    // write .timestamp data file
+    std::string timestamp = header.get("osmosis_replication_timestamp");
+    if (timestamp.empty())
+    {
+        timestamp = "n/a";
+    }
+    util::Log() << "timestamp: " << timestamp;
+
+    storage::io::FileWriter timestamp_file(config.timestamp_file_name,
+                                           storage::io::FileWriter::HasNoFingerprint);
+
+    timestamp_file.WriteFrom(timestamp.c_str(), timestamp.length());
```

Comment:
Hmm, true it seems like this is not being used. This should be ticketed and done in a separate PR I think.